api/launchpad/v1/launchpad.proto

```diff
@@ -0,0 +1,161 @@
+
+service LaunchpadService {
+  rpc UploadMetadatas(UploadMetadatasRequest) returns (UploadMetadatasResponse);
+  rpc CalculateCollectionMerkleRoot(CalculateCollectionMerkleRootRequest) returns (CalculateCollectionMerkleRootResponse);
+  rpc TokenMetadata(TokenMetadataRequest) returns (TokenMetadataResponse);
+  rpc LaunchpadProjectsByCreator(LaunchpadProjectsByCreatorRequest) returns (LaunchpadProjectsByCreatorResponse);
+  rpc LaunchpadProjects(LaunchpadProjectsRequest) returns (LaunchpadProjectsResponse);
+  rpc LaunchpadProjectById(LaunchpadProjectByIdRequest) returns (LaunchpadProjectByIdResponse);
+  rpc LaunchpadProjectsCounts(LaunchpadProjectsCountsRequest) returns (LaunchpadProjectsCountsResponse);
+  rpc ProposeApproveProject(ProposeApproveProjectRequest) returns (ProposeApproveProjectResponse);
+}
+
+enum Sort {
+  SORT_UNSPECIFIED = 0;
+  SORT_COLLECTION_NAME = 1;
+}
+
+enum SortDirection {
+  SORT_DIRECTION_UNSPECIFIED = 0;
+  SORT_DIRECTION_ASCENDING = 1;
+  SORT_DIRECTION_DESCENDING = 2;
+}
+
+enum Status {
+  STATUS_UNSPECIFIED = 0;
+  STATUS_INCOMPLETE = 1;
+  STATUS_COMPLETE = 2;
+  STATUS_REVIEWING = 3;
+  STATUS_CONFIRMED = 4;
+}
+
+// -------------------------------
+
+message LaunchpadProjectsByCreatorRequest {
+  string creator_id = 1;
+  string network_id = 2;
+  int32 limit = 3;
+  int32 offset = 4;
+  Sort sort = 5;
+  SortDirection sort_direction = 6;
+  optional Status status = 7;
+}
+
+message LaunchpadProjectsByCreatorResponse {
+  repeated LaunchpadProject projects = 1;
+}
+
+message LaunchpadProjectsRequest {
+  string network_id = 1;
+  int32 limit = 2;
+  int32 offset = 3;
+  Sort sort = 4;
+  SortDirection sort_direction = 5;
+  optional Status status = 6;
+}
+
+message LaunchpadProjectsResponse {
+  repeated LaunchpadProject projects = 1;
+}
+
+message LaunchpadProjectByIdRequest {
+  string network_id = 1;
+  string project_id = 2;
+}
+
+message LaunchpadProjectByIdResponse {
+  LaunchpadProject project = 1;
+}
+
+message UploadMetadatasRequest {
+  string sender = 1;
+  string network_id = 2;
+  string project_id = 3;
+  repeated Metadata metadatas = 4;
+  optional string pinata_jwt = 5;
+}
+
+message UploadMetadatasResponse {
+  string merkle_root = 1;
+}
+
+message CalculateCollectionMerkleRootRequest {
+  string sender = 1;
+  repeated Metadata metadatas = 2;
+}
+
+message CalculateCollectionMerkleRootResponse {
+  string merkle_root = 1;
+}
+
+message TokenMetadataRequest {
+  string sender = 1;
+  string network_id = 2;
+  string project_id = 3;
+  uint32 token_id = 4;
+}
+
+message TokenMetadataResponse {
+  string merkle_root = 1;
+  Metadata metadata = 2;
+  repeated string merkle_proof = 3;
+}
+
+message LaunchpadProjectsCountsRequest {
+  string network_id = 1;
+}
+
+message LaunchpadProjectsCountsResponse {
+  repeated StatusCount status_counts = 1;
+}
+
+message ProposeApproveProjectRequest {
+  string sender = 1;
+  string network_id = 2;
+  string project_id = 3;
+  string proposal_id = 4;

```

Comment:
if this is about approving a project request what about: ``ApproveProjectRequestProposal``

Reply:
It's about updating the DB after creating+approving a proposal.
`ProposeApproveProjectRequest` is called after the proposal has been created, it's not used to create a proposal, we create the proposal on front using DA0 contract.

The flow in front is:
- Make a proposal using DA0 DA0 (onchain)
- Get the proposal id
- Vote "yes" using DA0 DA0 (onchain)
- Trigger ProposeApproveProject to update db (indexer)

This flow is here: https://github.com/TERITORI/teritori-dapp/blob/feat/cosmwasm-launchpad/packages/hooks/launchpad/useProposeApproveProject.ts#L30

Reply:
this should be the responsibility of the indexer

Reply:
this should be the responsibility of an indexer handle, not triggered by an api call

Reply:
> this should be the responsibility of an indexer handler, not triggered by an api call
It makes sense

Reply:
> this should be the responsibility of an indexer handler, not triggered by an api call

I will need some instructions to make this refacto.
Which wasm action to listen ? This one ? https://github.com/TERITORI/teritori-dapp/blob/feat-launchpad-backend/go/internal/indexerhandler/handle.go#L325-L328
Should we use that as entry point from front ? https://github.com/TERITORI/teritori-dapp/blob/feat/cosmwasm-launchpad/packages/hooks/launchpad/useProposeApproveProject.ts#L30

Reply:
it would be the execute action on the dao contract here, but it's already handled and we need to listen to the submsg here really https://github.com/TERITORI/teritori-dapp/blob/feat-launchpad-backend/go/internal/indexerhandler/dao.go#L264
I'm not sure what is the correct action on the launchpad contract though, it would be the one proposed.

Reply:
Hm ok, but we need to handle things **before** "deploy_collection".
Either we listen for "propose" (DAO) (Then trigger the launchpad indexer stuff)
Either we trigger ProposeApproveProject (Launchpad backend) (Then trigger the DAO stuff)
No ?